src/app/services/overlap.py

```diff
@@ -0,0 +1,387 @@
+        Latitude of point 1 in decimal degrees.
+    lon1 : float
+        Longitude of point 1 in decimal degrees.
+    lat2 : float
+        Latitude of point 2 in decimal degrees.
+    lon2 : float
+        Longitude of point 2 in decimal degrees.
+
+    Returns
+    -------
+    float
+        Distance between the two points in kilometers.
+    """
+    r_earth = 6371.0
+    dlat = radians(lat2 - lat1)
+    dlon = radians(lon2 - lon1)
+    a = sin(dlat / 2) ** 2 + cos(radians(lat1)) * cos(radians(lat2)) * sin(dlon / 2) ** 2
+    c = 2 * atan2(sqrt(a), sqrt(1 - a))
+    return r_earth * c
+
+
+def get_centroid_latlon(geom: BaseGeometry) -> Tuple[float, float]:
+    """
+    Compute the geographic coordinates of the centroid of a given geometry.
+
+    Parameters
+    ----------
+    geom : BaseGeometry
+        Geometry in EPSG:3857 (Web Mercator projection).
+
+    Returns
+    -------
+    tuple[float, float]
+        Latitude and longitude of the centroid in EPSG:4326.
+    """
+    centroid = geom.centroid
+    transformer = pyproj.Transformer.from_crs("EPSG:3857", "EPSG:4326", always_xy=True)
+    lon, lat = transformer.transform(centroid.x, centroid.y)
+    return float(lat), float(lon)
+
+
+def _build_cone_polygon(
+    lat: float,
+    lon: float,
+    azimuth: float,
+    opening_angle: float,
+    dist_km: float,
+    r_min_km: float,
+    resolution: int = 36,
+) -> Polygon:
+    """
+    Build a cone sector polygon on the sphere then return it in geographic coordinates.
+
+    Parameters
+    ----------
+    lat : float
+        Camera latitude.
+    lon : float
+        Camera longitude.
+    azimuth : float
+        Cone central azimuth in degrees.
+    opening_angle : float
+        Full opening angle in degrees.
+    dist_km : float
+        Outer radius in kilometers.
+    r_min_km : float
+        Inner radius in kilometers.
+    resolution : int
+        Number of points to sample the arc.
+
+    Returns
+    -------
+    shapely.geometry.Polygon
+        Cone polygon in EPSG:4326 coordinates.
+    """
+    half_angle = opening_angle / 2.0
+    angles = np.linspace(azimuth - half_angle, azimuth + half_angle, resolution)
+
+    # Outer arc points
+    outer_arc = [geodesic(kilometers=dist_km).destination((lat, lon), float(az % 360)) for az in angles]
+    outer_points = [(p.longitude, p.latitude) for p in outer_arc]
+
+    if r_min_km > 0:
+        # Inner arc points, walk reversed so ring orientation stays valid
+        inner_arc = [geodesic(kilometers=r_min_km).destination((lat, lon), float(az % 360)) for az in reversed(angles)]
+        inner_points = [(p.longitude, p.latitude) for p in inner_arc]
+        # Outer ring with a hole for the inner radius
+        return Polygon(outer_points + inner_points, holes=[inner_points]).buffer(0)
+    # Triangle like sector with apex at camera position
+    return Polygon([(lon, lat), *outer_points]).buffer(0)
+
+
+def _project_polygon_to_3857(polygon: Polygon) -> Polygon:
```

Comment:
do you this the name of the function should be more explicit ?
```suggestion
def _project_polygon_from_4326_to_3857(polygon: Polygon) -> Polygon:
```